src/main/java/com/mathworks/ci/actions/RunMatlabAction.java

```diff
@@ -0,0 +1,81 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import com.mathworks.ci.BuildArtifactAction;
+import com.mathworks.ci.BuildConsoleAnnotator;
+import com.mathworks.ci.MatlabBuilderConstants;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+import hudson.FilePath;
+import hudson.model.Run;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class RunMatlabAction {
+    MatlabCommandRunner runner;
+    BuildConsoleAnnotator annotator;
+    String actionID;
+
+    public String getActionID(){
+        return this.actionID;
+    }
+
+    public RunMatlabAction(MatlabCommandRunner runner) {
+        this.runner = runner;
+    }
+
+    public RunMatlabAction(MatlabCommandRunner runner, BuildConsoleAnnotator annotator) {
+        this.runner = runner;
+        this.actionID = RandomStringUtils.randomAlphanumeric(8);
+        this.annotator = annotator;
+    }
+
+    public void copyBuildPluginsToTemp() throws IOException, InterruptedException {
+        // Copy plugins and override default plugins function
+        runner.copyFileToTempFolder(MatlabBuilderConstants.DEFAULT_PLUGIN, MatlabBuilderConstants.DEFAULT_PLUGIN);
+        runner.copyFileToTempFolder(MatlabBuilderConstants.BUILD_REPORT_PLUGIN, MatlabBuilderConstants.BUILD_REPORT_PLUGIN);
+        runner.copyFileToTempFolder(MatlabBuilderConstants.TASK_RUN_PROGRESS_PLUGIN, MatlabBuilderConstants.TASK_RUN_PROGRESS_PLUGIN);
+    }
+
+    public void setBuildEnvVars() throws IOException, InterruptedException {
+        // Set environment variable
+        runner.addEnvironmentVariable(
+                "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE",
+                "ciplugins.jenkins.getDefaultPlugins");
+        runner.addEnvironmentVariable("MW_BUILD_PLUGIN_ACTION_ID",this.getActionID());
+        runner.addEnvironmentVariable(
+                "MW_MATLAB_TEMP_FOLDER",
+                runner.getTempFolder().toString());
+    }
+
+    public void moveArtifactToBuildRoot(Run<?,?> build, String artifactBaseName) {

```

Comment:
I think that we should maybe try and figure out a name that also includes the fact that we're cleaning up the temp directory, because I think that's potentially even more important than moving the build artifact. I think it would help readability if it was clear to devs that were looking at the other files that this is a clean up function.

What do you think about `tearDownActions` or something in that vein? Then we could also use it in `RunMatlabTestsAction.java`

Reply:
I like the idea of having a `teardownActions()` method. I'll figure out some of the Exception related things and update the implementation